awx/main/access.py

```diff
@@ -1125,6 +1125,46 @@ def can_delete(self, obj):
         return self.can_change(obj, None)
 
 
+class CredentialInputSourceAccess(BaseAccess):
+    '''
+    I can see a CredentialInputSource when:
+     - I can see the associated target_credential
+    I can create/change a CredentialInputSource when:
+     - I'm an admin of the associated target_credential
+     - I have use access to the associated source credential
+    I can delete a CredentialInputSource when:
+     - I'm an admin of the associated target_credential
+    '''
+
+    model = CredentialInputSource
```

Comment:
you could add a `select_related` or `prefetch_related` for the `target_credential` and `source_credential`. That will make the listings of these go from 🐢 to 🐇